tutorials/how_to/span_representations.md

```diff
@@ -0,0 +1,180 @@
+`SpanExtractor` takes a sequence tensor of shape `(batch_size, sentence_length, embedding_size)`
+and some indices of shape `(batch_size, num_spans, 2)` and return the relevant indices into
+the sequence as a tensor of shape `(batch_size, num_spans, encoded_size)`.
+
+The simplest `SpanExtractor` is the `EndpointSpanExtractor`, which epresents spans as a 
+combination of the embeddings of their endpoints.
+
+```
+import torch
+from torch.autograd import Variable
+from allennlp.modules.span_extractors import EndpointSpanExtractor
+sequence_tensor = Variable(torch.randn([2, 5, 7]))
+# concatentate start and end points together to form our representation.
+extractor = EndpointSpanExtractor(input_dim7, combination="x,y")
+
+# Typically these would come from a SpanField,
+# rather than being created directly.
+indices = Variable(torch.LongTensor([[[1, 3],
+                                      [2, 4]],
+                                     [[0, 2],
+                                      [3, 4]]]))
+
+# We concatenated the representations for the start and end of
+# the span, so the embedded span size is 2 * embedding_size.
+# shape (batch_size, num_spans, 2 * embedding_size).
+span_representations = extractor(sequence_tensor, indices)
+assert list(span_representations.size()) == [2, 2, 14]
+```
+
+There are other types of Span Extractors - for instance, the `SelfAttentiveSpanExtractor`,
+which computes span representations by generating an unnormalized attention score for each
+word in the sentence. Spans representations are then computed with respect to these
+scores by normalising the attention scores for words inside the span.
+
+
+## Scoring and Pruning Spans
+
+Span based representations have been effective for modelling/approximating structured
+prediction problems - however, many models which leverage this type of representation
+also involve some kind of _span enumeration_ (i.e considering all possible spans in a 
+sentence/document). For a given sentence of length n, there are n^2 spans. In itself, 
+this is not too problematic, but for instance, the co-reference model in AllenNLP 
+compares _pairs_ of spans - meaning that naively we consider n^4 spans, with potential
+document lengths of upwards of 3000 tokens.
+
+In order to solve this problem, we need to be able to _prune_ spans as we go inside our
+model. There are several ways to do this:
+
+### Heuristically prune spans in your DatasetReader.
+
+We have added a utility method for enumerating all spans in a sentence, but excluding
+those which fulfil some condition based on the input text or any Spacy `Token` attribute.
+For instance, for co-reference, all spans which are mentions (spans which are 
+co-referent with _something_) never start or end with punctuation, because of the way
+the Onotonotes 5.0 dataset was created. This means that we can exclude any span which
+starts or ends with punctuation using a very simple python function:
+
```

Comment:
You might mention in here also that you probably don't want cross-sentence spans, which cuts down on complexity, also.